icetea/vm/js/Context.js

```diff
@@ -40,23 +40,25 @@ exports.contextForWrite = (tx, block, stateTable, {address, fname, fparams}) =>
     return Object.freeze(ctx);
 }
 
-exports.contextForView = (stateTable, address, name, params) => {
+exports.contextForView = (stateTable, address, name, params, options) => {
     const msg = new Proxy({name, params, callType: "view"}, {
         get(target, prop) {
-            if (Object.keys(msg).includes(prop) && !["name", "params", "callType"].includes(prop)) {
+            if (Object.keys(msg).includes(prop) && !["name", "params", "callType", "sender"].includes(prop)) {
                 throw new Error ("Cannot access msg." + prop + " when calling a @view function")
             }
             return Reflect.get(target, prop);
         }
     })
 
+    msg.sender = options.from;
+
     const state = _.cloneDeep(stateTable[address].state || {});
     const balance = stateTable[address].balance || 0;
 
     const ctx = {
         address,
         balance,
-        getEnv: () => ({msg}),
+        getEnv: () => ({msg, block: options.block, now: options.block.timestamp}),
```

Comment:
Why should we allow contract access to block in a "view" function. Because the call is not a transaction and not "mined", what's the block then?

Reply:
In solidity, if call `now` or `block.number` in view function, it will return the latest block information.